In kernel 2.6.26 and older CONFIG_IWL4965 was build as an extra
module, but now it is directly included in the iwlagn.

CONFIG_IWL4965 has to be set to y, to build correctly.

--- a/drivers/net/wireless/iwlwifi/Makefile	2009-10-06 12:57:49.000000000 -0400
+++ b/drivers/net/wireless/iwlwifi/Makefile	2009-10-06 12:57:50.000000000 -0400
@@ -12,7 +12,7 @@
 obj-$(CONFIG_IWLAGN)	+= iwlagn.o
 iwlagn-objs		:= iwl-agn.o iwl-agn-rs.o iwl-agn-led.o
 
-iwlagn-$(CONFIG_IWL4965) += iwl-4965.o
+iwlagn-$(CONFIG_COMPAT_IWL4965) += iwl-4965.o
 iwlagn-$(CONFIG_IWL5000) += iwl-5000.o
 iwlagn-$(CONFIG_IWL5000) += iwl-6000.o
 iwlagn-$(CONFIG_IWL5000) += iwl-1000.o
--- a/drivers/net/wireless/iwlwifi/iwl-agn.c	2009-11-02 08:07:33.000000000 -0800
+++ b/drivers/net/wireless/iwlwifi/iwl-agn.c	2009-11-02 08:07:33.000000000 -0800
@@ -3323,10 +3323,10 @@
 
 /* Hardware specific file defines the PCI IDs table for that hardware module */
 static struct pci_device_id iwl_hw_card_ids[] = {
-#ifdef CONFIG_IWL4965
+#ifdef CONFIG_COMPAT_IWL4965
 	{IWL_PCI_DEVICE(0x4229, PCI_ANY_ID, iwl4965_agn_cfg)},
 	{IWL_PCI_DEVICE(0x4230, PCI_ANY_ID, iwl4965_agn_cfg)},
-#endif /* CONFIG_IWL4965 */
+#endif /* CONFIG_COMPAT_IWL4965 */
 #ifdef CONFIG_IWL5000
 	{IWL_PCI_DEVICE(0x4232, 0x1205, iwl5100_bg_cfg)},
 	{IWL_PCI_DEVICE(0x4232, 0x1305, iwl5100_bg_cfg)},
